packages/ui/src/components/va-dropdown/VaDropdown.vue

```diff
@@ -1,88 +1,64 @@
+import { DropdownOffsetProp } from './types'
+import { useDropdown } from './hooks/useDropdown'
 
 export default defineComponent({
   name: 'VaDropdown',
-
-  inheritAttrs: false,
-
   props: {
     ...createStatefulProps(Boolean, true),
-    ...useComponentPresetProp,
-    ...usePlacementAliasesProps,
-    disabled: { type: Boolean },
-    readonly: { type: Boolean },
     anchorSelector: { type: String, default: '' },
     innerAnchorSelector: { type: String, default: '' },
-    /** Viewport where dropdown will be rendered if preventOverflow is true. Autoplacement will be calculated relative to `target` */
-    target: { type: [String, Object] as PropType<MaybeHTMLElementOrSelector>, default: undefined },
-    /** Element where dropdown content will be rendered. */
-    teleport: { type: [String, Object] as PropType<MaybeHTMLElementOrSelector>, default: undefined },
-    preventOverflow: { type: Boolean, default: false },
-    keepAnchorWidth: { type: Boolean, default: false },
-    isContentHoverable: { type: Boolean, default: true },
-    closeOnContentClick: { type: Boolean, default: true },
+    trigger: {
+      type: String as PropType<'click' | 'right-click' | 'hover' | 'dblclick' | 'none'>,
+      default: 'click',
+      validator: (value: string) => ['click', 'right-click', 'hover', 'dblclick', 'none'].includes(value),
+    },
+    disabled: { type: Boolean },
+    readonly: { type: Boolean },
     closeOnClickOutside: { type: Boolean, default: true },
     closeOnAnchorClick: { type: Boolean, default: true },
+    closeOnContentClick: { type: Boolean, default: true },
     hoverOverTimeout: { type: Number, default: 30 },
     hoverOutTimeout: { type: Number, default: 200 },
+    isContentHoverable: { type: Boolean, default: true },
+    placement: { type: String as PropType<Placement | 'auto'>, default: 'bottom' },
     offset: { type: [Array, Number] as PropType<DropdownOffsetProp>, default: 0 },
-    stickToEdges: { type: Boolean, default: false },
-    autoPlacement: { type: Boolean, default: true },
+    keepAnchorWidth: { type: Boolean, default: false },
     cursor: { type: Boolean, default: false },
-    trigger: {
-      type: String as PropType<'click' | 'right-click' | 'hover' | 'dblclick' | 'none'>,
-      default: 'click',
-      validator: (value: string) => ['click', 'right-click', 'hover', 'dblclick', 'none'].includes(value),
-    },
+    autoPlacement: { type: Boolean, default: true },
+    stickToEdges: { type: Boolean, default: false },
+    /** Viewport where dropdown will be rendered if preventOverflow is true. Autoplacement will be calculated relative to `target` */
+    target: { type: [String, Object] as PropType<MaybeHTMLElementOrSelector>, default: undefined },
+    /** Element where dropdown content will be rendered. */
+    teleport: { type: [String, Object] as PropType<MaybeHTMLElementOrSelector>, default: undefined },
+    preventOverflow: { type: Boolean, default: false },
```

Comment:
Let's turn it on by default and maybe remove this prop, because it doesn't make any sense to not prevent overflow...